server/src/main/java/io/spine/server/entity/storage/ColumnReader.java

```diff
@@ -85,32 +112,52 @@ static ColumnReader forClass(Class<? extends Entity> entityClass) {
      * <p>If the check for correctness fails, throws {@link IllegalStateException}.
      *
      * @return a {@code Collection} of {@link EntityColumn} corresponded to entity class
-     * @throws IllegalStateException
-     *         if entity column definitions are incorrect
+     * @throws IllegalStateException if entity column definitions are incorrect
      */
-    @SuppressWarnings("ConstantConditions")
     Collection<EntityColumn> readColumns() {
-        BeanInfo entityDescriptor;
-        try {
-            entityDescriptor = Introspector.getBeanInfo(entityClass);
-        } catch (IntrospectionException e) {
-            throw new IllegalStateException(e);
-        }
+        Set<EntityColumn> columns = gatherColumnsFromProperties();
+        Set<EntityColumn> booleanWrapperColumns = gatherBooleanWrapperColumns();
+        columns.addAll(booleanWrapperColumns);
+        checkRepeatedColumnNames(columns);
+        return columns;
+    }
 
-        PropertyDescriptor[] propertyDescriptors = entityDescriptor.getPropertyDescriptors();
-        ImmutableSet<EntityColumn> columns = Arrays
-                .stream(propertyDescriptors)
+    /**
+     * Gathers entity properties and creates the entity columns where appropriate.
+     *
+     * <p>The entity properties search is based on the Java Bean
+     * <a href="https://download.oracle.com/otndocs/jcp/7224-javabeans-1.01-fr-spec-oth-JSpec/">
+     * specification</a>.
+     */
+    private Set<EntityColumn> gatherColumnsFromProperties() {
```

Comment:
Can't we join both of `gather...` methods into one?

And also, I would rename the method to something like `scanColumns`.